src/config/scrivitoSites.ts

```diff
@@ -0,0 +1,76 @@
+import { Obj, currentSiteId, load, navigateTo } from 'scrivito'
+
+const location = typeof window !== 'undefined' ? window.location : undefined
+
+export function baseUrlForSite(siteId: string): string | undefined {
+  const tenant = getTenantFromEnv()
+  if (!location || !tenant) return
+
+  const urlParts = [location.origin]
+
+  // Multitenancy mode
+  if (!import.meta.env.SCRIVITO_TENANT) urlParts.push(tenant)
+
+  const language = Obj.onSite(siteId).root()?.language()
+  if (language) urlParts.push(language)
+
+  return urlParts.join('/')
+}
+
+export function siteForUrl(
+  url: string,
+): { baseUrl: string; siteId: string } | undefined {
+  const language = /\b\/([0-9a-f]{32}\/)?(?<lang>[a-z]{2})([?/]|$)/.exec(url)
+    ?.groups?.lang
```

Comment:
Are language codes always 2 chars long?

Reply:
Simple ISO language codes: yes.

Reply:
So the question is: Does Scrivito always use "Simple ISO language codes" for the "_language" attribute?

Reply:
Non-custom languages are two-letter. The UI accepts any custom code as long as it consists of letters and dashes. In order to support this, we'd need to get all languages first and check them individually instead of guessing from the URL.

Another assumption the code currently makes is that there is only one website with language versions. The UI allows to create multiple sites with multiple versions though.

Reply:
> Another assumption the code currently makes is that there is only one website with language versions. The UI allows to create multiple sites with multiple versions though.

We could add a validation to the `Homepage` obj class, that there are not two site roots with the same language. If someone would like to create multiple sites with the same language, the code would need to be adjusted anyway.